lib/JsonParser.js

```diff
@@ -12,7 +12,7 @@ class JsonParser {
 	}
 
 	parse(source, state) {
-		const data = JSON.parse(source);
+		const data = JSON.parse(source.replace("\ufeff", ""));
```

Comment:
Maybe only replace it at the start of the source

Reply:
It causes an error anywhere in the file outside of double quotes

Reply:
`replace(/^\ufeff/, "")`

This don't scan the whole file for this char. It's more performant. And makes more sense logically.